File: src/main/java/forpleuvoir/hiirosakura/client/gui/event/WidgetListEvent.java
package forpleuvoir.hiirosakura.client.gui.event;

import fi.dy.masa.malilib.gui.interfaces.ISelectionListener;
import fi.dy.masa.malilib.gui.widgets.WidgetListBase;
import forpleuvoir.hiirosakura.client.config.HiiroSakuraDatas;
import forpleuvoir.hiirosakura.client.feature.event.base.EventSubscriberBase;

import java.util.Collection;

/**
 * @author forpleuvoir
 * <p>#project_name hiirosakura
 * <p>#package forpleuvoir.hiirosakura.client.gui.event
 * <p>#class_name WidgetListEvent
 * <p>#create_time 2021-07-28 18:24
 */
public class WidgetListEvent extends WidgetListBase<EventSubscriberBase, WidgetEventEntry> {

    private final String eventType;

    public WidgetListEvent(int x, int y, int width, int height, String eventType,
                           ISelectionListener<EventSubscriberBase> selectionListener
    ) {
        super(x, y, width, height, selectionListener);
        this.browserEntryHeight = 22;
        this.eventType = eventType;
    }

    @Override
    protected Collection<EventSubscriberBase> getAllEntries() {
        if (eventType.equals(EventScreen.ALL))
            return HiiroSakuraDatas.HIIRO_SAKURA_EVENTS.getAllEventSubscriberBase();
        else
            return HiiroSakuraDatas.HIIRO_SAKURA_EVENTS.getAllEventSubscriberBase(eventType);
    }

    @Override
    protected WidgetEventEntry createListEntryWidget(int x, int y, int listIndex, boolean isOdd,
                                                     EventSubscriberBase entry
    ) {
        return new WidgetEventEntry(x, y,
                this.browserEntryWidth,
                this.getBrowserEntryHeightFor(entry),
                isOdd,
                entry,
                listIndex,
                this
        );
    }
}

File: src/main/kotlin/forpleuvoir/hiirosakura/client/gui/event/WidgetListEvent.kt
package forpleuvoir.hiirosakura.client.gui.event

import fi.dy.masa.malilib.gui.interfaces.ISelectionListener
import fi.dy.masa.malilib.gui.widgets.WidgetListBase
import forpleuvoir.hiirosakura.client.config.HiiroSakuraData
import forpleuvoir.hiirosakura.client.feature.event.base.EventSubscriberBase

/**
 * @author forpleuvoir
 *
 * #project_name hiirosakura
 *
 * #package forpleuvoir.hiirosakura.client.gui.event
 *
 * #class_name WidgetListEvent
 *
 * #create_time 2021-07-28 18:24
 */
class WidgetListEvent(
	x: Int, y: Int, width: Int, height: Int, private val eventType: String?,
	selectionListener: ISelectionListener<EventSubscriberBase?>?
) : WidgetListBase<EventSubscriberBase, WidgetEventEntry>(x, y, width, height, selectionListener) {
	override fun getAllEntries(): Collection<EventSubscriberBase> {
		return if (eventType == EventScreen.ALL) HiiroSakuraData.HIIRO_SAKURA_EVENTS.allEventSubscriberBase else HiiroSakuraData.HIIRO_SAKURA_EVENTS.getAllEventSubscriberBase(
			eventType
		)
	}

	override fun createListEntryWidget(
		x: Int, y: Int, listIndex: Int, isOdd: Boolean,
		entry: EventSubscriberBase
	): WidgetEventEntry {
		return WidgetEventEntry(
			x, y,
			browserEntryWidth,
			getBrowserEntryHeightFor(entry),
			isOdd,
			entry,
			listIndex,
			this
		)
	}

	init {
		browserEntryHeight = 22
	}
}